### PR TITLE
AWS Lambda SDK: Fix dev mode support and improve its testing

### PR DESCRIPTION
**Pull request by medikoo**
Fix handling of telemetry requests with dev mode enabled

Additionally:
- Introduce dummy external extension to take payloads from internal, to enable better testing of dev mode.
- Update benchmarks to cover also external extension case and ensure performance check is run in CI

### CODE REVIEW

File: node/packages/aws-lambda-sdk/instrument/index.js [hunk 1]
```diff
@@ -161,6 +160,7 @@ module.exports = (originalHandler, options = {}) => {
       reportTrace();
       flushSpans();
       await Promise.all(serverlessSdk._deferredTelemetryRequests);
+      serverlessSdk._deferredTelemetryRequests.length = 0;
```
Comment (Danwakeem) [inline node/packages/aws-lambda-sdk/instrument/index.js:163]: TIL that you can set an array length to 0 to clear the array contents 🤯

File: node/packages/aws-lambda-sdk/test/fixtures/dev-mode-extension/extensions/dummy-dev-extension.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env node
```
Comment (Danwakeem) [inline node/packages/aws-lambda-sdk/test/fixtures/dev-mode-extension/extensions/dummy-dev-extension.js:1]: I am curious why we wouldn't just use the actual dev mode extension?

**Review by Danwakeem — COMMENTED** [review]
Looking good 👍 just one question about the mock external extension

File: node/packages/aws-lambda-sdk/test/fixtures/dev-mode-extension/extensions/dummy-dev-extension.js (hunk 2 above)
Comment (medikoo) [reply node/packages/aws-lambda-sdk/test/fixtures/dev-mode-extension/extensions/dummy-dev-extension.js:1]: This is a no-op extension, through which we can isolate any side-effects as created by the internal extension. It's meant to expose issues in internal extension only. If I would introduce real external extension here its issues may create false positives.

Still, totally we should also have full integration test, that confirms that external extension sends expected payloads  but I think it's more in the scope of the external extension package (although it be nice to have such tests run also on any changes in this extension)

**Review by Danwakeem — APPROVED** [review]
This is looking good 👍

File: node/packages/aws-lambda-sdk/test/fixtures/dev-mode-extension/extensions/dummy-dev-extension.js (hunk 2 above)
Comment (Danwakeem) [reply node/packages/aws-lambda-sdk/test/fixtures/dev-mode-extension/extensions/dummy-dev-extension.js:1]: Okay yeah this makes sense 👍

**Pull request merged**